packages/autoloader/src/class-hook-manager.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/* HEADER */ // phpcs:ignore
+
+/**
+ * Allows the latest autoloader to register hooks that can be removed when the autoloader is reset.
+ */
+class Hook_Manager {
+
+	/**
+	 * An array containing all of the hooks that we've registered.
+	 *
+	 * @var array
+	 */
+	private $registered_hooks;
+
+	/**
+	 * The constructor.
+	 */
+	public function __construct() {
+		$this->registered_hooks = array();
+	}
+
+	/**
+	 * Adds an action to WordPress and registers it internally.
+	 *
+	 * @param string   $tag           The name of the action which is hooked.
+	 * @param callable $callable      The function to call.
+	 * @param int      $priority      Used to specify the priority of the action.
+	 * @param int      $accepted_args Used to specify the number of arguments the callable accepts.
+	 */
+	public function add_action( $tag, $callable, $priority = 10, $accepted_args = 1 ) {
+		$this->registered_hooks[ $tag ][] = array(
+			'priority' => $priority,
+			'callable' => $callable,
+		);
+
+		add_action( $tag, $callable, $priority, $accepted_args );
+	}
+
+	/**
+	 * Adds a filter to WordPress and registers it internally.
+	 *
+	 * @param string   $tag           The name of the filter which is hooked.
+	 * @param callable $callable      The function to call.
+	 * @param int      $priority      Used to specify the priority of the filter.
+	 * @param int      $accepted_args Used to specify the number of arguments the callable accepts.
+	 */
+	public function add_filter( $tag, $callable, $priority = 10, $accepted_args = 1 ) {
```

Comment:
Since the `Hook_Manager` is intended to be a shared global I thought it would be wise to add the rest of its API to avoid potential bugs if filter support was added. To be honest I don't see a use-case for filters in the autoloader given its utility but I didn't want to rule out the possibility since it's a small function.

Reply:
We would normally remove unused code, but I think it's ok to include it in this case.